From 29492dc9e1985e5fa54ee9474150347ca2e665b6 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Tue, 19 Jul 2011 18:10:54 +0800
Subject: [PATCH] Revert "rk29: vpu_service: fix bug when __cancel_delayed_work
 and delay work run at the same time"

This reverts commit c41dd263406cad12ec15d39986a530be1d23ad11.
---
 arch/arm/mach-rk29/vpu_service.c | 6 +-----
 1 file changed, 1 insertion(+), 5 deletions(-)

diff --git a/arch/arm/mach-rk29/vpu_service.c b/arch/arm/mach-rk29/vpu_service.c
index 35c64a99815d..b0d08b29967d 100644
--- a/arch/arm/mach-rk29/vpu_service.c
+++ b/arch/arm/mach-rk29/vpu_service.c
@@ -391,11 +391,7 @@ static void try_set_reg(void)
        		    ((VPU_PP  == reg->type) && (NULL == service.reg_pproc)) ||
        		    ((VPU_ENC == reg->type) && (NULL == service.reg_codec))) {
 			reg_from_wait_to_run(reg);
-			if (in_interrupt()) {
-				__cancel_delayed_work(&vpu_service_power_off_work);
-			} else {
-				cancel_delayed_work(&vpu_service_power_off_work);
-			}
+			__cancel_delayed_work(&vpu_service_power_off_work);
 			vpu_service_power_on();
 			reg_copy_to_hw(reg);
 		}
-- 
2.35.3

